File: grid.py
import pyglet
cell_image = pyglet.image.load('cell.png')
class Grid():
    def __init__(self,x_size,y_size,no_of_cells,cell_width):
        self.x_size = x_size
        self.y_size = y_size
        self.no_of_cells = no_of_cells
        ## TODO:  <07-04-20, no_of_cells is not relevant since cell no will be dependent on x_size and y_size#
        self.cell_width = cell_width
        self.cells = []
        self.main_batch = pyglet.graphics.Batch()

    def initialize_grid(self):
        for y in range(1, self.y_size+1):
            for x in range(1, self.x_size+1):
                self.cells.append(pyglet.sprite.Sprite(cell_image,x*self.cell_width,y*self.cell_width,batch = self.main_batch))
                print("made cell pos x:{x} y:{y} ".format(x=x*self.cell_width,y = y*self.cell_width))
    def draw(self):
        self.main_batch.draw()
        

File: grid.js
class Grid {
  constructor(x_size, y_size, cell_width) {
    this.x_size = x_size;
    this.y_size = y_size;
    this.cell_width = cell_width;
    this.cell_count = 0;
    this.gap = 0;
  }
  initialize_grid() {
    this.y_size = this.y_size / this.cell_width;
    this.x_size = this.x_size / this.cell_width;
    this.draw_width = this.cell_width - this.gap
    this.cells = [];
    for (var y = 0; y < this.y_size; y++) {
      for (var x = 0; x < this.x_size; x++) {
        this.cells.push(new Cell(x * this.cell_width, y * this.cell_width, this.cell_width, 0));
        console.log(`made cell ${this.cell_count} at pos x:${x*this.cell_width} y:${y*this.cell_width}`) //This is a debug log
        this.cell_count++;
      }
    }
    this.runningConfig()
  }
  runningConfig() {
    // TODO:  <13-04-20> add more configurations //
    //glider configuration config
    let reference_cell = this.cells.filter((cell) => {
      return (cell.y === (this.y_size / 2) * this.cell_width && cell.x === (this.x_size / 2) * this.cell_width)
    })
    console.log({
      cell_to_refer_to: reference_cell
    })
    this.cells.forEach((cell) => {
      if (cell.x === reference_cell[0].x + this.cell_width && (cell.y === reference_cell[0].y + this.cell_width)) {
        cell.working_state = 1;
        console.log("changed cell state 1")
      }
      if (cell.x === reference_cell[0].x - this.cell_width && cell.y === reference_cell[0].y + this.cell_width * 2) {
        cell.working_state = 1;
        console.log("changed cell state 2")
      }
      if (cell.x === reference_cell[0].x && cell.y === reference_cell[0].y + this.cell_width * 2) {
        cell.working_state = 1;
        console.log("changed cell state 3")
      }
      if (cell.x === reference_cell[0].x + this.cell_width && cell.y === reference_cell[0].y + this.cell_width * 2) {
        cell.working_state = 1;
        console.log("changed cell state 4")
      }
      if (cell.x === reference_cell[0].x && cell.y === reference_cell[0].y) {
        cell.working_state = 1;
        console.log("changed cell state 5")
      }
    })

  }
  observeAndSimulate() {
    this.cells.forEach((cell) => {
      cell.update(this.cells)
    })
  }
  updateStates() {
    this.cells.forEach((cell) => {
      cell.working_state = cell.new_state
    })
  }
}